numpy/_core/src/common/get_attr_string.h

```diff
@@ -59,6 +60,11 @@ PyArray_LookupSpecial(
         return 0;
     }
 
+    /* We do not need to check for special attributes on array descriptor types */
```

Comment:
I don't think this is a good idea - `PyArray_LookupSpecial` is used also to look for `__array_ufunc__` and `__array_function__` and I don't think those paths should be made slower for `result_type`. 

p.s. Separately, is this actually used with `result_type`?

Reply:
I will remove this one, the speed gain is not major. This path is being used if `result_type` as called with `dtype` arguments (not with exact ndarray arguments), I think because the `np._result_type` is wrapped by `array_function_from_c_func_and_dispatcher`.

Reply:
It should be OK to exclude dtypes in the dispatcher, but probably better in a separate PR.